src/structures/GuildChannel.js

```diff
@@ -414,18 +414,19 @@ class GuildChannel extends Channel {
    * @param {string} [options.reason] Reason for cloning this channel
    * @returns {Promise<GuildChannel>}
    */
-  clone({ name = this.name, withPermissions = true, withTopic = true, nsfw, parent, bitrate, userLimit, reason } = {}) {
-    const options = {
-      overwrites: withPermissions ? this.permissionOverwrites : [],
-      nsfw: typeof nsfw === 'undefined' ? this.nsfw : nsfw,
-      parent: parent || this.parent,
-      bitrate: bitrate || this.bitrate,
-      userLimit: userLimit || this.userLimit,
-      reason,
-      type: this.type,
-    };
-    return this.guild.channels.create(name, options)
-      .then(channel => withTopic ? channel.setTopic(this.topic) : channel);
+  clone(options = {}) {
+    Util.mergeDefault({
+      name: this.name,
+      withPermissions: true,
```

Comment:
Looks indeed better now, but `withPermissions` is not a valid option when creating a channel.
Not sure about a nice way to fix this, I guess you could do something like this after the mergeDefault
`if (options.withPermissions) options.overwrites = this.permissionOverwites;`

Reply:
Oh, sure. I'll apply that change in a moment, thanks for the suggestion 👍

Reply:
That comment was actually to the commit before, the current one is imo good as well